.github/workflows/tests.yml

```diff
@@ -0,0 +1,48 @@
+name: tests
+
+# This action is triggered:
+# 1. when someone creates a pull request for a merge to the main branch
+# 2. when changes are merged into the main branch (via a pull request)
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
+
+    # we support Linux and macOS
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+
+    # Steps for running tests and analysis.
+    steps:
+      - name: Checking out repository (${{ matrix.os }})
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          submodules: recursive
+
+      - name: Setting up Python 3.10 (${{ matrix.os }})
+        uses: actions/setup-python@v5.1.1
+        with:
+          python-version: "3.10"
```

Comment:
any reason for using py 3.10 instead of the current version, 3.12?

GHA change relatively quickly so it might be easier to use `actions/setup-python@v5` so you don't have to keep on updating the version